### PR TITLE
fix for thermostat reporting null target temp

### LINKED ISSUE #2647: Nest Thermostat Eco mode shows 'null' in Lovelace Temperature Card
**Home Assistant release with the issue:**
0.86.4

**Last working Home Assistant release (if known):**
Unknown - 0.86 is when I started to use Lovelace 

**Browser and Operating System:**

macOS 10.14.3, Safari 12.0.3

**Description of problem:**
When the Nest Thermostat is in the Away/Eco mode the Lovelace temperature card shows the temperature as null. It seems like this is being taken from the entity's `temperature` attribute which is set to null - perhaps a null check should be added and the value is not displayed in this scenario?

Imgur
Imgur

**Javascript errors shown in the web inspector (if applicable):**
N/A

**Additional information:**
N/A

### PR DESCRIPTION
**Pull request by iantrich**
Fixes https://github.com/home-assistant/home-assistant-polymer/issues/2647

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 1]
```diff
@@ -287,7 +287,9 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
   }
 
   private _updateSetTemp(value: string): void {
-    this.shadowRoot!.querySelector("#set-temperature")!.innerHTML = value;
+    if (value !== "null") {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: It's not the string `"null"`, but the value `null`.

Also, we should still set a value, so just set it to empty.

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 2]
```diff
@@ -287,7 +287,9 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
...
+      this.shadowRoot!.querySelector("#set-temperature")!.innerHTML = value;
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]:
```suggestion
      this.shadowRoot!.querySelector("#set-temperature")!.innerHTML = value || "";
```

**Comment by balloob** [comment]
Actually, we should research why we get `null` to begin with.

File: src/panels/lovelace/cards/hui-thermostat-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: It forces it to a string, doesn't it? https://github.com/home-assistant/home-assistant-polymer/blob/0e47dace2204f5a43c24470655b9534550dc263e/src/panels/lovelace/cards/hui-thermostat-card.ts#L283

Comment (balloob) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: oh you're right, that's kinda gross that it happens there and not inside this method.

Comment (iantrich) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: agreed, but uiValue is of type string, which is probably the reason for it. Do you want me to refactor it?

Comment (balloob) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: Yeah, have it pass `null` without converting. In this method, if it's `null`, set the content to an empty string.

**Review by balloob — APPROVED** [review]
Beautiful.

**Pull request merged**